File: DockerLogDriver/src/main/java/de/derteufelqwe/driver/endpoints/VolumeDriverRemoveEP.java
package de.derteufelqwe.driver.endpoints;

import de.derteufelqwe.driver.messages.VolumeDriver;

import java.io.Serializable;

public class VolumeDriverRemoveEP extends Endpoint<VolumeDriver.RRemove, VolumeDriver.Remove> {

    public VolumeDriverRemoveEP(String data) {
        super(data);
    }

    @Override
    protected VolumeDriver.Remove process(VolumeDriver.RRemove request) {
        return new VolumeDriver.Remove();
    }

    @Override
    protected Class<? extends Serializable> getRequestType() {
        return VolumeDriver.RRemove.class;
    }

    @Override
    protected Class<? extends Serializable> getResponseType() {
        return VolumeDriver.Remove.class;
    }

}

File: DockerLogDriver/src/main/java/de/derteufelqwe/driver/endpoints/VolumeDriverRemoveEP.kt
package de.derteufelqwe.driver.endpoints

import de.derteufelqwe.driver.messages.VolumeDriver
import java.io.Serializable

class VolumeDriverRemoveEP(data: String?) : Endpoint<VolumeDriver.RRemove, VolumeDriver.Remove>(data) {

    override fun process(request: VolumeDriver.RRemove): VolumeDriver.Remove {
        return VolumeDriver.Remove()
    }

    override fun getRequestType(): Class<out Serializable?> {
        return VolumeDriver.RRemove::class.java
    }

    override fun getResponseType(): Class<out Serializable?> {
        return VolumeDriver.Remove::class.java
    }
}